lib/smart_enum/yaml_store.rb

```diff
@@ -12,10 +18,17 @@ def register_values_from_file!
         raise "Cannot infer data file for anonymous class"
       end
 
-      filename = "#{SmartEnum::Utilities.tableize(self.name)}.yml"
-      file_path = File.join(SmartEnum::YamlStore.data_root, filename)
-      values = YAML.load_file(file_path)
-      register_values(values, self, detect_sti_types: true)
+      basename = SmartEnum::Utilities.tableize(self.name)
+      dirname = File.join(SmartEnum::YamlStore.data_root, basename)
+      files = if Dir.exists?(dirname)
```

Comment:
If `dirname` and `#{basename}.yml` both exist, this will ignore the latter, which isn't the most intuitive behavior.  I would either raise or grab it as well.